src/App.tsx

```diff
@@ -15,49 +18,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, handleSortType] = useState(SortType.NONE);
+  const [isReversed, handleReverse] = useState(false);
+  const sortByAlphabet = () => handleSortType(SortType.ALPABET);
+  const sortByLength = () => handleSortType(SortType.LENGTH);
+  const reverse = () => handleReverse(!isReversed);
```

Comment:
Method's name should start with a verb (so you could easily tell what your method actually do)
For example 'getReverse'
Same in method 'reset'